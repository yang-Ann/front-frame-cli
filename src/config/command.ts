const config = {
	/**
   * 命令配置
	 * 键值需要对应 {@link TemplateParamsType} 的键(除了 git)
	 */
	execCommandOption: {
		frame: [
			{ name: "Vue", checked: true },
			{ name: "React" },
			{ name: "Native" },
		],
		language: [
			{ name: "TypeScript", checked: true },
			{ name: "JavaScript" }
		],
		packages: (frame: FrameType) => getPackagesOptionByFrame(frame),
	},
}


// 根据框架返回对应的可以选择的 package
type PackageListType = Array<{ name: string, checked?: boolean }>;
const getPackagesOptionByFrame = (frame: FrameType): PackageListType => {

	let choicesPackages: PackageListType = [];
	// 通用 package
	const common = [
		{ name: "Eslint", checked: true },
		{ name: "Axios" },
		{ name: "Prettier" },
		{ name: "Lodash" },
		{ name: "Dayjs" },
		{ name: "Husky" },
		{ name: "Commitlint" },
	];


	switch (frame) {
		case "Vue":
			choicesPackages = [
				{ name: "Element-Plus", checked: true },
				{ name: "Sass", checked: true },
				{ name: "Pinia" },
				{ name: "Vue-router" },
				{ name: "Vueuse" },
				{ name: "JSX(TSX)" },
				{ name: "Tailwind CSS" },
			];
			break;
		case "React":
			choicesPackages = [
				{ name: "Ant-Design", checked: true },
				{ name: "Reducx" },
				{ name: "React-router" },
				{ name: "React-use" },
				{ name: "Tailwind CSS" },
			];
			break;
		case "Native":
			choicesPackages = [
				{ name: "Vitest", checked: true },
				{ name: "Fs-extra", checked: true },
				{ name: "Rollup" },
				{ name: "Ora" },
				{ name: "Chalk" },
				{ name: "Log-symbols" },
				{ name: "Puppeteer-core" },
				// { name: "Commander" },
				// { name: "Inquirer" },
			];
			break;
		default:
			console.log("未知的类型 -->", frame);
			break;
	}

	// 排序
	return [
		...choicesPackages,
		...common
	].sort((a, b) => a.name.localeCompare(b.name));
};


export const { execCommandOption } = config;
export {
  getPackagesOptionByFrame
};