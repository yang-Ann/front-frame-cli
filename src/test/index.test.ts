// https://cn.vitest.dev/guide/
import { test, expect, describe } from "vitest";
import { resolve } from "node:path";
import {
  objKeySort,
  strAsAscll,
  getExtByLang,
  walkdirOpator,
  isChildObject,
} from "../utils/exports.js";
import pkg from "../../package.json";
import { getPackagesOptionByFrame } from "../config/command.js";

describe("utils", () => {
  test("strAsAscll", async () => {
    const res = await strAsAscll(pkg.name);
    expect(res).toMatchSnapshot("打印 ASCLL");
  }, 3000);

  test("getExtByLang", () => {
    expect(getExtByLang("JavaScript")).toMatchInlineSnapshot('".js"');
    expect(getExtByLang("JavaScript", true)).toMatchInlineSnapshot('".jsx"');

    expect(getExtByLang("TypeScript")).toMatchInlineSnapshot('".ts"');
    expect(getExtByLang("TypeScript", true)).toMatchInlineSnapshot('".tsx"');
  });

  test("walkdirOpator", async () => {
    const path = resolve(__dirname, "../../assets/template/Vue/TypeScript");
    const RE = /\\/g;
    const files = await walkdirOpator(path, p => {
      return p.endsWith(".ejs") ? p.replace(RE, "/") : false;
    });
    expect(files).toMatchSnapshot("walkdirOpator 读取 ejs 文件信息");
  });

  test("isChildObject", () => {
    const obj = {
      Eslint: {
        // ...
      }
    };

    expect(isChildObject(obj)).toMatchInlineSnapshot('true');
  });
});


describe("config", () => {

  test("objKeySort", () => {
    const obj = {
      a: "1",
      c: "2",
      b: "3",
      "@types/node": "4"
    }

    expect(objKeySort(obj, "ASC")).toMatchInlineSnapshot(`
      {
        "@types/node": "4",
        "a": "1",
        "b": "3",
        "c": "2",
      }
    `);
  });


  test("getPackagesOptionByFrame", () => {
    expect(getPackagesOptionByFrame("Vue")).toMatchSnapshot("Vue 可选依赖");

    expect(getPackagesOptionByFrame("React")).toMatchSnapshot("React 可选依赖");

    expect(getPackagesOptionByFrame("Native")).toMatchSnapshot("Native 可选依赖");
  });
});