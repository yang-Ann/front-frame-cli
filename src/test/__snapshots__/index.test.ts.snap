// Vitest Snapshot v1

exports[`config > getPackagesOptionByFrame > Native 可选依赖 1`] = `
[
  {
    "name": "Axios",
  },
  {
    "name": "Chalk",
  },
  {
    "name": "Commitlint",
  },
  {
    "name": "Dayjs",
  },
  {
    "checked": true,
    "name": "Eslint",
  },
  {
    "checked": true,
    "name": "Fs-extra",
  },
  {
    "name": "Husky",
  },
  {
    "name": "Lodash",
  },
  {
    "name": "Log-symbols",
  },
  {
    "name": "Ora",
  },
  {
    "name": "Playwright",
  },
  {
    "name": "Prettier",
  },
  {
    "name": "Puppeteer-core",
  },
  {
    "name": "Rollup",
  },
  {
    "checked": true,
    "name": "Vitest",
  },
]
`;

exports[`config > getPackagesOptionByFrame > React 可选依赖 1`] = `
[
  {
    "checked": true,
    "name": "Ant-Design",
  },
  {
    "name": "Axios",
  },
  {
    "name": "Commitlint",
  },
  {
    "name": "Dayjs",
  },
  {
    "checked": true,
    "name": "Eslint",
  },
  {
    "name": "Husky",
  },
  {
    "name": "Lodash",
  },
  {
    "name": "Prettier",
  },
  {
    "name": "React-router",
  },
  {
    "name": "React-use",
  },
  {
    "name": "Reducx",
  },
  {
    "name": "Tailwind CSS",
  },
]
`;

exports[`config > getPackagesOptionByFrame > Vue 可选依赖 1`] = `
[
  {
    "name": "Axios",
  },
  {
    "name": "Commitlint",
  },
  {
    "name": "Dayjs",
  },
  {
    "checked": true,
    "name": "Element-Plus",
  },
  {
    "checked": true,
    "name": "Eslint",
  },
  {
    "name": "Husky",
  },
  {
    "name": "JSX(TSX)",
  },
  {
    "name": "Lodash",
  },
  {
    "name": "Pinia",
  },
  {
    "name": "Prettier",
  },
  {
    "checked": true,
    "name": "Sass",
  },
  {
    "name": "Tailwind CSS",
  },
  {
    "name": "Vue-router",
  },
  {
    "name": "Vueuse",
  },
]
`;

exports[`utils > strAsAscll > 打印 ASCLL 1`] = `
"    ___                          _              ___                                                  __    _   
  .' ..]                        / |_          .' ..]                                                [  |  (_)  
 _| |_  _ .--.   .--.   _ .--. \`| |-'______  _| |_  _ .--.  ,--.   _ .--..--.  .---.  ______  .---.  | |  __   
'-| |-'[ \`/'\`\\\\]/ .'\`\\\\ \\\\[ \`.-. | | | |______|'-| |-'[ \`/'\`\\\\]\`'_\\\\ : [ \`.-. .-. |/ /__\\\\\\\\|______|/ /'\`\\\\] | | [  |  
  | |   | |    | \\\\__. | | | | | | |,          | |   | |    // | |, | | | | | || \\\\__.,        | \\\\__.  | |  | |  
 [___] [___]    '.__.' [___||__]\\\\__/         [___] [___]   \\\\'-;__/[___||__||__]'.__.'        '.___.'[___][___] 
                                                                                                               "
`;
